### PR TITLE
[WIP] Promote existing directory

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promotes the following E2E tests to Conformance:
test/e2e/storage/testsuites/subpath.go: "should support existing directory"

**Special notes for your reviewer**:
Part of Umbrella Issue #78747
[Test grid results](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=subPath%20should%20support%20existing%20directory).

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```
/area conformance
/area test
@kubernetes/sig-storage-pr-reviews
@kubernetes/sig-architecture-pr-reviews
@kubernetes/cncf-conformance-wg
/assign @hh

### CODE REVIEW

File: test/conformance/testdata/conformance.txt [hunk 1]
```diff
@@ -219,3 +219,4 @@ test/e2e/storage/subpath.go: "should support subpaths with configmap pod"
 test/e2e/storage/subpath.go: "should support subpaths with configmap pod with mountPath of existing file"
 test/e2e/storage/subpath.go: "should support subpaths with downward pod"
 test/e2e/storage/subpath.go: "should support subpaths with projected pod"
+test/e2e/storage/testsuites/subpath.go: "should support existing directory"
```
Comment (hh) [inline test/conformance/testdata/conformance.txt]: Let's update the PR title to better match the full test name in testgrid... maybe without the tags.

File: test/e2e/storage/testsuites/subpath.go [hunk 2]
```diff
@@ -179,6 +179,11 @@ func (s *subPathTestSuite) defineTests(driver TestDriver, pattern testpatterns.T
 		testReadFile(f, l.filePathInVolume, l.pod, 1)
 	})
 
+	/*
+	   Release : v1.16
+	   Testname: Create directory in existing path
+	   Description: MUST be able to create directory in existing path
+	*/
 	ginkgo.It("should support existing directory", func() {
```
Comment (hh) [inline test/e2e/storage/testsuites/subpath.go]: Will need to update to framework.ConformanceIt

File: test/e2e/storage/testsuites/subpath.go [hunk 3]
```diff
@@ -179,6 +179,11 @@ func (s *subPathTestSuite) defineTests(driver TestDriver, pattern testpatterns.T
 		testReadFile(f, l.filePathInVolume, l.pod, 1)
 	})
 
+	/*
```
Comment (hh) [inline test/e2e/storage/testsuites/subpath.go]: While the test is pretty straight forward the fact that it's a storage test may cause issues.
I would assume any cloud provider would have to support creating a directory and ensuring it's still there. :smile_cat:

**Comment by spiffxp** [comment]
https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&include-filter-by-regex=should%20support%20existing%20directory

So this test is exercised by a _lot_ of different storage drivers.  We can't expect a conformant cluster to have/support all of these.  Are we trying to call the behavior conformant? Are we trying to say it's conformant only for a certain driver?

eg: given `[sig-storage] In-tree Volumes [Driver: emptydir] [Testpattern: Inline-volume (default fs)] subPath should support existing directory` which makes more sense
- `[sig-storage] In-tree Volumes [Conformance] [Driver: emptydir] [Testpattern: Inline-volume (default fs)] subPath should support existing directory`
- `[sig-storage] In-tree Volumes [Driver: emptydir] [Testpattern: Inline-volume (default fs)] subPath should support existing directory [Conformance]`

**Comment by msau42** [comment]
I agree for conformance tests I think [Driver: emptydir] makes sense.

Unfortunately these test cases are written to be driver-agnostic, so you can run the same test case against any storage driver.  This doesn't really fit well with the way that conformance tests are tagged and tracked.  If we want to promote a test case like this, then we'll need to think about how we can conditionally tag a test case.

File: test/conformance/testdata/conformance.txt (hunk 1 above)
Comment (devaips) [reply test/conformance/testdata/conformance.txt]: That would actually cause some tests to fail, as the Conformance tooling expects the string to exactly match the test name.

**Comment by devaips** [comment]
As @msau42 [said ](https://github.com/kubernetes/kubernetes/pull/79962#issuecomment-510245280) we need to be able to conditionally tag a test case before being able to safely promote tests like this .

/close

**Comment by BobyMCbobs** [comment]
As @msau42 said we need to be able to conditionally tag a test case before being able to safely promote tests like this. Closing until then.

/close

**Pull request closed without merging**